dataactbroker/handlers/fileHandler.py

```diff
@@ -406,19 +418,22 @@ def uploadFile(self):
 
     def generateD1File(self):
         """ Initiates the generation of D1 """
+        jobDb = self.interfaces.jobDb
         requestDict = RequestDictionary(self.request)
 
         if not (requestDict.exists("submission_id") and requestDict.exists("start") and requestDict.exists("end")):
             exc = ResponseException("Generate D1 Files route requires submission_id, start, and end", StatusCode.CLIENT_ERROR)
-            return JsonResponse.create(exc, exc.status)
+            return JsonResponse.error(exc, exc.status)
 
         submission_id = requestDict.getValue("submission_id")
         start_date = requestDict.getValue("start")
         end_date = requestDict.getValue("end")
-
+        job = jobDb.getJobBySubmissionFileTypeAndJobType(submission_id, self.EXTERNAL_FILE_TYPE_MAP["D1"], "file_upload")
+        job.start_date = datetime.strptime(start_date,"%m/%d/%Y").date()
+        job.end_date = datetime.strptime(end_date,"%m/%d/%Y").date()
```

Comment:
These lines are storing the data in the jobs table to be returned in the later call to /check_generation_status/.  448 stores other info in the jobs table for the same purpose.  My thinking in putting these lines here was to be closer to where start_date and end_date are created so it's clearer what is being stored.  That said, I can see how it could be clearer to have all the job-related lines in one place, so I'll move these down to 448.